crates/cxx-qt-gen/src/generator/cpp/property/mod.rs

```diff
@@ -96,7 +129,41 @@ mod tests {
             struct MyStruct;
         };
 
-        let generated = setup_generated(&mut input).unwrap();
+        // let generated = setup_generated(&mut input).unwrap();
+
+        let property = ParsedQProperty::parse(input.attrs.remove(0)).unwrap();
+
+        let properties = vec![property];
+
+        let qobject_idents = create_qobjectname();
+
+        // Prototyping, this test need properly rewriting
+        let module: ItemMod = parse_quote! {
+            #[cxx_qt::bridge]
+            mod ffi {
+                extern "RustQt" {
+                    #[qobject]
+                    type MyObject = super::MyObjectRust;
+                }
+
+                unsafe extern "RustQt" {
+                    fn mySetter(self: Pin<&mut MyObject>, value: i32);
+                }
+            }
+        };
+        let parser = Parser::from(module).unwrap();
+        let structures = Structures::new(&parser.cxx_qt_data).unwrap();
+
+        let structured_qobject = structures.qobjects.get(0).unwrap();
+
+        let type_names = TypeNames::mock();
+        let generated = generate_cpp_properties(
+            &properties,
+            &qobject_idents,
+            &type_names,
+            &structured_qobject,
+        )
+        .unwrap();
```

Comment:
@LeonMatthesKDAB any suggestions on how we can nicely mock the `Structures` from our other discussions ? Eg should we be building the whole parser and using parse_quote etc here. Or should we manually have ways to build up the Structure and ParsedMethod etc.

Reply:
Hm, we should be mocking it, yes. I'll need to think a bit about this after the workshop.

Reply:
Have introduced functions just to remove that block from tests, but still could be improved I think

Reply:
Hm, the new mocking functions look alright, so maybe that's just the way to go :)

Reply:
so sorted then?

Reply:
I think so.